crates/router/src/core/payment_methods/cards.rs

```diff
@@ -660,6 +681,89 @@ pub async fn get_client_secret_or_add_payment_method(
     }
 }
 
+#[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
+#[instrument(skip_all)]
+pub async fn get_client_secret_or_add_payment_method(
```

Comment:
What is the client_secret part here?, the function implementation says only about adding a payment_method.

Reply:
In case the payment_method_data is present in body, we take the data and vault directly (S2S call).
In case it is not present, we create a payment_method entry along with a client_secret and send back the response. This client_secret can be used to vault payment_method using SDK (the `/save` endpoint)